client_plugin/vmdk_plugin/main.go

```diff
@@ -17,185 +17,44 @@ package main
 	"github.com/docker/go-plugins-helpers/volume"
-	"github.com/natefinch/lumberjack"
 	"github.com/vmware/docker-volume-vsphere/client_plugin/drivers/photon"
 	"github.com/vmware/docker-volume-vsphere/client_plugin/drivers/vmdk"
 	"github.com/vmware/docker-volume-vsphere/client_plugin/utils/config"
+	"github.com/vmware/docker-volume-vsphere/client_plugin/utils/plugin_server"
 )
 
-// PluginServer responds to HTTP requests from Docker.
-type PluginServer interface {
-	// Init initializes the server.
-	Init()
-	// Destroy destroys the server.
-	Destroy()
-}
-
-// init log with passed logLevel (and get config from configFile if it's present)
-// returns True if using defaults,  False if using config file
-func logInit(logLevel *string, logFile *string, configFile *string) bool {
-	usingConfigDefaults := false
-	c, err := config.Load(*configFile)
-	if err != nil {
-		if os.IsNotExist(err) {
-			usingConfigDefaults = true // no .conf file, so using defaults
-			c = config.Config{}
-			config.SetDefaults(&c)
-		} else {
-			panic(fmt.Sprintf("Failed to load config file %s: %v",
-				*configFile, err))
-		}
-	}
-
-	path := c.LogPath
-	if logFile != nil {
-		path = *logFile
-	}
-	log.SetOutput(&lumberjack.Logger{
-		Filename: path,
-		MaxSize:  c.MaxLogSizeMb,  // megabytes
-		MaxAge:   c.MaxLogAgeDays, // days
-	})
-
-	if *logLevel == "" {
-		*logLevel = c.LogLevel
-	}
-
-	level, err := log.ParseLevel(*logLevel)
-	if err != nil {
-		panic(fmt.Sprintf("Failed to parse log level: %v", err))
-	}
-
-	log.SetFormatter(new(VmwareFormatter))
-	log.SetLevel(level)
-
-	if usingConfigDefaults {
-		log.Info("No config file found. Using defaults.")
-	}
-	return usingConfigDefaults
-}
-
 // main for docker-volume-vsphere
 // Parses flags, initializes and mounts refcounters and finally initializes the server.
 func main() {
 	var driver volume.Driver
 
-	// Get options from ENV (where available), and from command line.
-	// ENV takes precedence, so we can modify it in Docker plugin install
-	logEnv := os.Getenv("VDVS_LOG_LEVEL")
-	logLevel := &logEnv
-	if *logLevel == "" {
-		logLevel = flag.String("log_level", "", "Logging Level")
-	}
-	configFile := flag.String("config", config.DefaultConfigPath, "Configuration file path")
-	driverName := flag.String("driver", "", "Volume driver")
-
-	// Photon driver options
-	targetURL := flag.String("target", "", "Photon controller URL")
-	projectID := flag.String("project", "", "Project ID of the docker host")
-	vmID := flag.String("host", "", "ID of docker host")
-
-	// vSphere driver options
-	port := flag.Int("port", defaultPort, "Default port to connect to ESX service")
-	useMockEsx := flag.Bool("mock_esx", false, "Mock the ESX service")
-
-	flag.Parse()
-
-	logInit(logLevel, nil, configFile)
-
-	// Load the configuration if one was provided.
-	c, err := config.Load(*configFile)
+	cfg, err := config.InitConfig(config.DefaultConfigPath, config.DefaultLogPath,
+		config.VSphereDriver, config.VSphereDriver)
```

Comment:
We need to pass parameters when we add the new shared plugin.
For now we only have the vmdk plugin so it looks unnecessary.
But this is required when we have the new shared plugin in the following PRs.

It will become the following for vmdk plugin:
`
cfg, err := config.InitConfig(config.DefaultVMDKPluginConfigPath, config.DefaultVMDKPluginLogPath, config.VSphereDriver, config.VSphereDriver)`

And for shared plugin:
`
cfg, err := config.InitConfig(config.DefaultSharedPluginConfigPath, config.DefaultSharedPluginLogPath, config.SharedDriver, config.SharedDriver)`